src/ba-rfcomm.c

```diff
@@ -1283,11 +1303,19 @@ static void *rfcomm_thread(struct ba_rfcomm *r) {
 			switch (rfcomm_recv_signal(r)) {
 #if ENABLE_MSBC
 			case BA_RFCOMM_SIGNAL_HFP_SET_CODEC_CVSD:
+				if ((r->sco->profile & BA_TRANSPORT_PROFILE_HFP_AG) && !(r->hfp_features & HFP_HF_FEAT_CODEC)) {
```

Comment:
I'm trying to test "Fix HFP codec selection logic" commit, and I wonder why the `rfcomm_finalize_codec_selection()` is sent only in case of AG here? I've just looked through all the "if" branches, in here in order to fix potential deadlocks due to error being returned by `rfcomm_set_hfp_codec()`, and I'm kind of inclined to loosen the condition here to just `if (!(r->hfp_features & HFP_HF_FEAT_CODEC))`. For codec negotiation all involved parties (AG and HF) have to support it. So it seems reasonable to skip sending `+BCS` command if either of them does not support `FEAT_CODEC`.

Reply:
After some tests I've decided to check `FEAT_CODEC` for both sides, i.e. AG and HF.